src/commons/withScrollEnabler.tsx

```diff
@@ -22,24 +22,27 @@ function withScrollEnabler<PROPS extends SupportedViews>(
 ): React.ComponentType<PROPS> {
   const ScrollEnabler = (props: PropTypes) => {
     const [scrollEnabled, setScrollEnabled] = useState(true);
-    const [contentSize, setContentSize] = useState(0);
-    const [layoutSize, setLayoutSize] = useState(0);
+    const contentSize = useRef(0);
+    const layoutSize = useRef(0);
 
-    useEffect(() => {
-      const isScrollEnabled = contentSize > layoutSize;
+    const checkScroll = useCallback(() => {
+      const isScrollEnabled = contentSize.current > layoutSize.current;
       if (isScrollEnabled !== scrollEnabled) {
         setScrollEnabled(isScrollEnabled);
       }
-    }, [contentSize, layoutSize]);
+    }, []);
```

Comment:
If you're using something inside the callback it should be part of the dependencies array of the useCallback
in this case you're using `scrollEnabled` state, so it should be included in the array.

Reply:
Done